### DISCUSSION TITLE
Destroying entity on collision and registry size question.

### DISCUSSION THREAD

**Discussion by rudiHammad** [skypjack/entt#1033] [Q&A] [answered]
hi,
I have 2 entities in my scene. They have a sprite component and movement component, and they are moving to collide with each other.
Here is my CollisionSystem.

`struct CollisionSystem
{
public:

    CollisionSystem() = default;

    void update(entt::registry& registry)
    {   
        
        auto view = registry.view<SpriteComponent>();
        ax::log("%i <", registry.size());
    
        for (auto i = view.begin(); i != view.end(); i++)
        {
            auto entA = *i;
            auto spriteCmpA = view[entA];  
            for (auto j = i; j != view.end(); j++)
            {
                auto entB = *j;
                auto spriteCmpB = view[entB];

                // Bypass if we are trying to test same entity
                if (*i == *j)
                    continue;

                // check the collision between entA and entB
                if (spriteCmpA.sprite->getBoundingBox().intersectsRect(spriteCmpB.sprite->getBoundingBox()))
                {
                    registry.destroy(entA);
                    registry.destroy(entB);
                    ax::log("%i <<", registry.size());
                }
            }
        }
    }
};`

I have printed the size before and after the collision with ax::log(___), which is axmol engine logger.
The printed result at the beginning is 2 as expected, but after destroying both entities it is still 2. Why is not zero since I destroyed both?
And in the next loop the game crashes with the error **expression: contains(entt)**.
I am a bit confused on both the registry size results and the crash. I guess are they related but I am not sure.
Is the logic of my collision system wrong?

Thanks.
R

**Comment by skypjack** [comment]
As from the documentation, deleting the entity currently iterated is allowed while deleting other entities is UB.
Imagine what could happen if you deleted an element in a vector during a range-for loop. That's more or less what you're doing here. 🙂

**Comment by rudiHammad** [comment]
yup, I see it now.

I solved by creating a small struct called DeadTag. And now, instead of destroying on collision, the system adds that tag. And another method in the gamescene  at the end of the update loop checks wich entities has the DeadTag component, and destroys it.
I saw similar things in other code repose so I guess that is the right way?

One more thing, I have been looking the documentation and searching in the discussions and google to find out how to check if an entity has a specific component and all I saw is a get functions. For instance  `auto& spriteComp = registry.get<SpriteComponent>(entity);`
This is not exactly what I want, I am looking for a boolean return. I apologize if the answer is right in front of me in the  docs but I didn't see any. Plus, comming from python, c++ could be a bit overwhelming with all the syntax. Sometimes I peek Entt code I see something like this ` [[nodiscard]] decltype(auto) get([[maybe_unused]] const entity_type entt) ` and my head explodes.

Thx

**Reply by skypjack** [reply, marked as answer]
No worries. You can use `any_of` and `all_of` from the registry interface. Otherwise, you can invoke `contains` on a view or a storage type. 👍
